components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/subscription/registry/RegistrySubscriptionManager.java

```diff
@@ -71,71 +92,63 @@ public RegistrySubscriptionManager(String topicStoragePath, String indexStorageP
+     * {@inheritDoc}
      */
+    @Override
     public void addSubscription(Subscription subscription) throws EventBrokerException {
-
-
         try {
             UserRegistry userRegistry =
-                    this.registryService.getGovernanceSystemRegistry(EventBrokerHolder.getInstance().getTenantId());
-            String resourcePath = getResourcePath(subscription.getId(), subscription.getTopicName());
+                    this.registryService.getGovernanceSystemRegistry(EventBrokerHolder.getInstance()
+                                                                             .getTenantId());
+            String resourcePath =
+                    getResourcePath(subscription.getId(), subscription.getTopicName());
 
             Resource resource = userRegistry.newResource();
-            resource.setProperty(EventBrokerConstants.EB_RES_SUBSCRIPTION_URL, subscription.getEventSinkURL());
-            resource.setProperty(EventBrokerConstants.EB_RES_EVENT_DISPATCHER_NAME, subscription.getEventDispatcherName());
-
-            if (subscription.getExpires() != null){
-               resource.setProperty(EventBrokerConstants.EB_RES_EXPIRS, ConverterUtil.convertToString(subscription.getExpires()));
+            resource.setProperty(EventBrokerConstants.EB_RES_SUBSCRIPTION_URL, subscription
+                    .getEventSinkURL());
+            resource.setProperty(EventBrokerConstants.EB_RES_EVENT_DISPATCHER_NAME, subscription
+                    .getEventDispatcherName());
+
+            if (subscription.getExpires() != null) {
+                resource.setProperty(EventBrokerConstants.EB_RES_EXPIRS, ConverterUtil
+                        .convertToString(subscription
+                                                 .getExpires()));
             }
             resource.setProperty(EventBrokerConstants.EB_RES_OWNER, subscription.getOwner());
-            resource.setProperty(EventBrokerConstants.EB_RES_TOPIC_NAME, subscription.getTopicName());
-            resource.setProperty(EventBrokerConstants.EB_RES_CREATED_TIME, System.currentTimeMillis()+"");
-            resource.setProperty(EventBrokerConstants.EB_RES_MODE, JavaUtil.getSubscriptionMode(subscription.getTopicName()));
+            resource.setProperty(EventBrokerConstants.EB_RES_TOPIC_NAME, subscription
+                    .getTopicName());
+            resource.setProperty(EventBrokerConstants.EB_RES_CREATED_TIME,
+                                 System.currentTimeMillis() + "");
+            resource.setProperty(EventBrokerConstants.EB_RES_MODE, JavaUtil
+                    .getSubscriptionMode(subscription
+                                                 .getTopicName()));
 
             //set the other properties of the subscription.
             Map<String, String> properties = subscription.getProperties();
-            for (String key : properties.keySet()){
+            for (String key : properties.keySet()) {
```

Comment:
Could we get the EntrySet ? so that we don't need to do another lookup for retrival